src/lib/vm/vm.js

```diff
@@ -1581,12 +1581,37 @@ export default class VM {
           cacheOptions
         );
       },
+      signAndSendTransactions: (args) => {
+        return this.confirmTransactions(args);
+      },
       call: (...args) => {
         if (args.length === 1) {
           if (isObject(args[0])) {
-            return this.confirmTransactions([args[0]]);
+            return this.confirmTransactions([
+              {
+                receiverId: args[0].contractName,
+                actions: [{ type: "FunctionCall", params: args[0] }],
+              },
+            ]);
           } else if (isArray(args[0])) {
-            return this.confirmTransactions(args[0]);
+            let transactions = [];
+            args[0].forEach((action) => {
+              const lastReceiverId =
+                transactions[transactions.length - 1]?.receiverId;
+              if (lastReceiverId !== action.contractName) {
+                transactions.push({
+                  receiverId: action.contractName,
+                  actions: [{ type: "FunctionCall", params: action }],
+                });
+              } else {
+                transactions[transactions.length - 1].actions.push({
+                  type: "FunctionCall",
+                  params: action,
+                });
+              }
```

Comment:
You still need to limit the max gas amount per transaction. As 2FA contracts are not the issue, the max gas should be 300Tgas